colcon_bundle/verb/_overlay_utilities.py

```diff
@@ -127,36 +107,36 @@ def recursive_tar_gz_in_path(output_path, path):
     :param path: path to recursively collect all files and include in
     tar.gz. These will be included with path as the root of the archive.
     """
+    tar_path = Path(path)
     with tarfile.open(output_path, mode='w:gz', compresslevel=5) as tar:
         logger.info(
-            'Creating tar of {path}'.format(path=path))
-        for name in os.listdir(path):
-            some_path = os.path.join(path, name)
-            tar.add(some_path, arcname=os.path.basename(some_path))
+            'Creating tar of {path}'.format(path=str(tar_path)))
+        for name in tar_path.iterdir():
+            some_path = Path(tar_path) / name
+            tar.add(str(some_path), arcname=os.path.basename(str(some_path)))
 
 
-def _generate_template(template_name, script_name, context_vars):
+def _rendering_template(template_name: str,
+                        script_dest: Path,
+                        context_vars: dict):
     """
-    Generate setup.bash or setup.sh files from a template.
+    Render setup.bash or setup.sh files from template.
 
     This assumes the template is in the assets folder.
 
     :param template_name: Name of the template to be used
-    :param script_name: name of the script to be generated
+    :param script_dest: path of the script to be rendered
     :param context_vars: dictionary of values to be used for the variables in
     the template
     """
-    template_location = os.path.join(
-        os.path.dirname(os.path.realpath(__file__)), 'assets/')
+    template_location = Path(__file__).parent.absolute() / 'assets/'
```

Comment:
```suggestion
    template_location = Path(__file__).parent.absolute() / 'assets'
```

Building in path separators in the strings defeats the purpose of using a path lib